module/module.go

```diff
@@ -276,25 +279,11 @@ func (m *Module) Start(ctx context.Context) error {
 }
 
 // Close shuts down the module and grpc server.
-// TODO: Shut down the subscriptions.
 func (m *Module) Close(ctx context.Context) {
 	m.closeOnce.Do(func() {
+		m.cancelFn()
 		m.mu.Lock()
 		parent := m.parent
-		for name, r := range m.activeResourceStreams {
-			vcss, ok := m.streamSourceByName[name]
-			if !ok {
-				m.logger.CError(ctx, "Module Close unable to find resource", "name", name,
-					"in streamSourceByName", fmt.Sprintf("%#v", m.streamSourceByName))
-				continue
-			}
-			if err := vcss.Unsubscribe(ctx, r.subID); err != nil {
-				m.logger.CErrorw(ctx, "Unsubscribe", "name", name, "err", err, "subID", r.subID.String())
-			}
-			if err := m.pc.RemoveTrack(r.sender); err != nil {
-				m.logger.CErrorw(ctx, "RemoveTrack", "name", name, "err", err)
-			}
-		}
```

Comment:
This work is now done by Close() which causes the Subscrption.Context of all subscriptions to be cancelled, which then run the cleanup.

Reply:
This looks kosher to me. Just to state out loud something I checked -- it wasn't obvious to me if the existing calls to `Unsubscribe`/`RemoveTrack` blocked for background goroutines to complete.

I do [see](https://github.com/viamrobotics/rdk/pull/3792/files#diff-9a0a92d26bcd93d88bcfe6d94c11871002ddb11ea827c5f27c75c3238464d079R767) that `Module.AddStream` is adding `activeBackgroundWorkers`. Such that the `m.cancelFn` must propagate to those background workers and those workers must return before [the upcoming](https://github.com/viamrobotics/rdk/pull/3792/files#diff-9a0a92d26bcd93d88bcfe6d94c11871002ddb11ea827c5f27c75c3238464d079L313) `WaitGroup.Wait` returns.